content/running_bookwyrm/cli.md

```diff
@@ -0,0 +1,143 @@
+Title: Command Line Tool
+Date: 2021-11-06
+Order: 6
+
+Bookwyrm developers and instance managers can use the `bw-dev` script for common tasks. This can make your commands shorter, easier to remember, and harder mess up.
+
+Once you have installed Bookwyrm [in production](installing-in-production.html) or [in development](https://docs.joinbookwyrm.com/developer-environment.html#setting_up_the_developer_environment), you can run the script from the command line with `./bw-dev` followed by the subcommand you want to run.
+
+## Docker shortcuts
+
+### bash
+
+Open an interactive `bash` session inside the docker `web` container.
+
+### build
+
+Equivalent to `docker-compose build`.
+
+### dbshell
+
+Open an interactive Postgres database shell. I hope you know what you're doing.
+
+### run
+
+Run an arbitrary command in the `web` cointainer.
+
+Equivalent to `docker-compose run --rm --service-ports web`. 
+
+### rundb
+
+TODO: this command does not appear to actually exist!
+
+### runweb _args_
+
+Run an arbitrary command in the `web` container.
+
+Equivalent to `docker-compose run --rm web`.
+
+### shell
+
+Open an interactive Django shell inside the docker `web` container. You would use this if you want to run Django shell commands directly. For example when [making your initial user a superuser](installing-in-production.html#configure_your_instance).
+
+### up [args]
+
+Start or restart Docker containers. Equivalent to `docker-compose up --build [args]`
+
+## Managing the database
+
+### initdb
+
+Initialize a database.
+
+### makemigrations
+
+Runs Django's `makemigrations` command inside your Docker container. If you have changed the database structure in a development branch you will need to run this for your changes to have effect.
+
+### migrate
+
+Runs Django's `migrate` command inside your Docker container. You always need to run this after `makemigrations`.
+
+### resetdb
+
+Reset the database. **This command will delete your entire Bookwyrm database**. You should probably not run it in production! It is however useful for development servers. Note that `resetdb` will also initialise a fresh database and run all migrations, so you should delete any recent migration files you do not want to run, _before_ running `resetdb`.
+
+## Managing a Bookwyrm instance
+
+### collectstatic
+
+Migrate static assets to either a Docker container or to an S3-compatible "bucket", depending on the context.
+
+### generate_preview_images
+
+Generate preview images for site, users, and books. This can take a while if you have a large database.
+
+### populate_streams
+
+Re-populates Redis streams (user feeds). You will usually need to run this after `update`.
```

Comment:
In theory you shouldn't need to run this under any normal circumstances; just if you need to recover from a problem with the user feeds (like they got wiped out, or are incorrect)